spring-cloud-commons/src/main/java/org/springframework/cloud/client/loadbalancer/LoadBalancerAutoConfiguration.java

```diff
@@ -65,12 +67,35 @@ public void afterSingletonsInstantiated() {
 		};
 	}
 
+	@Bean
+	public LoadBalancerRequestTransformer uriReconstructingRequestTransformer() {
+		return new LoadBalancerRequestTransformer() {
+
+			@Override
+			public HttpRequest transformRequest(HttpRequest request, ServiceInstance instance,
+					LoadBalancerClient loadBalancer) {
+				return new UriReconstructingRequestWrapper(request, instance, loadBalancer);
```

Comment:
Why is this a transformer?  I would think Spring Cloud would create the initial request and then pass that request to any transformers that might be present.  Its not really transforming the request at all.